starlette/tasks.py

```diff
@@ -0,0 +1,30 @@
+import uuid
+import asyncio
+import functools
+
+
+class BackgroundTask:
+    def __init__(self, func, *args, **kwargs):
+        self.func = func
+        self.args = args
+        self.kwargs = kwargs
+
+    async def __call__(self):
+        if asyncio.iscoroutinefunction(self.func):
+            self._result = await asyncio.ensure_future(
+                self.func(*self.args, **self.kwargs)
+            )
+        else:
+            fn = functools.partial(self.func, *self.args, **self.kwargs)
+            loop = asyncio.get_event_loop()
+            self._result = await loop.run_in_executor(None, fn)
+
+    @property
+    def id(self):
+        if not hasattr(self, "_id"):
+            self._id = str(uuid.uuid4())
+        return self._id
+
+    @property
+    def result(self):
+        return getattr(self, "_result", None)
```

Comment:
I'm not convinced we should have an interface for `result` or `id`. Those seem superfluous.

Reply:
My thinking here was to support if a client needed access to the status or result of a task directly. Once a task is initiated the task.id could be appended to `content` and sent along with the response. 
So then a client could access like so..

`@app.route('/status/{task_id}')`

Might be too niche though

Reply:
That's a reasonable pattern, yup, I just don't think it's absolute enough that we should include it ourselves. Leave that part up to developers (or higher-level frameworks).

Reply:
I got you. I guess I'm just unsure how a future developer would access the return value of the task if we aren't setting it to a property on the class.

Reply:
I suppose generate the uuid and pass it to the background task as an argument.

Reply:
That we you can include it in the response content, and the background task has it available for updating any state it needs to.